lib/encode_repeating.rb

```diff
@@ -0,0 +1,6 @@
+# A method that updates the string by replacing consecutive repeating characters
+# with a number representing the frequency. The replacement is done only if the
+# string length will get reduced by this process.
+def encode_repeating(my_string)
+  my_string << "NOT IMPLEMENTED"
```

Comment:
How would you feel about doing `raise NotImplementedError` here instead?

Best practice I think is generally to avoid `puts` statements and other things that will print to the console for code being tested, because that output kind of gets randomly interspersed with the test runner's output.

Reply:
Great idea!